enhancements/installer/connected-onprem-assisted-installer.md

```diff
@@ -0,0 +1,298 @@
+---
+title: assisted-installer-on-prem
+authors:
+  - "@mhrivnak"
+reviewers:
+  - "@avishayt"
+  - "@hardys"
+  - "@dhellmann"
+  - "@beekhof"
+  - "@ronniel1"
+  - "@filanov"
+approvers:
+  - "@crawford"
+  - "@abhinavdahiya"
+  - "@eparis"
+creation-date: 2020-07-23
+last-updated: 2020-10-01
+status: implementable
+see-also:
+  - "/enhancements/installer/connected-assisted-installer.md"  
+---
+
+# Assisted Installer On-Prem
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The assisted installer, [previously proposed and
+accepted](connected-assisted-installer.md), includes a service running on
+cloud.redhat.com to which agents connect and wait for orders on how to
+A) provision the host they are running on, and B) join or form a cluster.
+
+This enhancement proposes to enable a user to run that "assistant" service
+on-premise, inside their own network and on their own infrastructure, for
+one-time-use to create a new cluster.
+
+## Motivation
+
+Many users will not want a service that can provision hosts inside their
+network to be run by a third party outside of their network. Reasons can
+include trust, control, and reproducibility. Those users will prefer to run the
+service on-premise.
+
+### Goals
+
+* Users can run the entire assisted install workflow with software that runs inside their network.
+
+### Non-Goals
+
+* Disconnected and air-gapped environments are not included in this enhancement.
+* Adding Nodes on day 2 will be addressed separately.
+* Utilizing a local container registry for mirrored content will be addressed in a future enhancement.
```

Comment:
Should we list that running the assisted services on the bootstrap host (e.g one of the controlplane nodes) is not yet in scope, and may be addressed via a future enhancement?

I think that will likely be a future requirement for customers that want this functionality, but don't want the constraint of providing an additional node (same complaint we've got re baremetal IPI, although obviously in this case the network requirements and method of install/boot proposed are somewhat simpler).

Reply:
That's definitely on the roadmap, but not for this enhancement, so I'll add it as a non-goal. Good suggestion.